xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererDX.cpp

```diff
@@ -213,17 +213,35 @@ COverlayImageDX::COverlayImageDX(CDVDOverlayImage* o)
     Load(rgba.data(), o->width, o->height, o->width * 4);
   }
 
-  if (o->source_width && o->source_height)
+  if (o->source_width > 0 && o->source_height > 0)
   {
-    float center_x = (0.5f * o->width + o->x) / o->source_width;
-    float center_y = (0.5f * o->height + o->y) / o->source_height;
-
-    m_align = ALIGN_VIDEO;
     m_pos = POSITION_RELATIVE;
-    m_x = center_x;
-    m_y = center_y;
-    m_width = static_cast<float>(o->width) / o->source_width;
-    m_height = static_cast<float>(o->height) / o->source_height;
+
+    m_x = (0.5f * o->width + o->x) / o->source_width;
+    m_y = (0.5f * o->height + o->y) / o->source_height;
+
+    if ((o->source_height == static_cast<int>(rSource.Height()) ||
+         (static_cast<int>(rSource.Height()) % o->source_height == 0)) &&
+        (o->source_width == static_cast<int>(rSource.Width()) ||
+         (static_cast<int>(rSource.Width()) % o->source_width == 0)))
```

Comment:
This code is hard to understand what do due excess of parentheses 😓

Can you please change to something like:

```c++
    int sh = static_cast<int>(rSource.Height());
    int sw = static_cast<int>(rSource.Width());

    if ((o->source_height == sh || sh % o->source_height == 0) &&
        (o->source_width == sw || sw % o->source_width == 0))
    {
```

Reply:
yes done